go/blob/gcs.go

```diff
@@ -30,21 +38,21 @@ func NewGCSBucket(ctx context.Context, bucket string, opts []option.ClientOption
 	}, nil
 }
 
-func (s *GCSBucket) NewReader(ctx context.Context, key string) (io.ReadCloser, error) {
-	r, err := s.client.Bucket(s.bucket).Object(key).NewReader(ctx)
+func (b *GCSBucket) NewReader(ctx context.Context, key string) (io.ReadCloser, error) {
+	r, err := b.client.Bucket(b.bucket).Object(key).NewReader(ctx)
 	if err != nil {
 		return nil, err
 	}
 
 	return r, nil
 }
 
-func (s *GCSBucket) NewWriter(ctx context.Context, key string, opts ...WriterOption) io.WriteCloser {
-	return newBlobWriteCloser(ctx, s.Upload, key, opts...)
+func (b *GCSBucket) NewWriter(ctx context.Context, key string, opts ...WriterOption) io.WriteCloser {
+	return b.client.Bucket(b.bucket).Object(key).NewWriter(ctx)
```

Comment:
A small change here - `NewWriter` from the GCS SDK already gives us a writer and manages the goroutine lifecycle internally, so we don't need to do that again with `newBlobWriteCloser`.